CMakeLists.txt

```diff
@@ -77,26 +77,9 @@ __sync_add_and_fetch(&a, 1);
 return 0;
 }" ARCH_SUPPORTS_64BIT_ATOMICS)
 
-CMAKE_PUSH_CHECK_STATE()
-SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -pthread")
-CHECK_C_SOURCE_COMPILES("
-#define _GNU_SOURCE
-#include <sched.h>
-#include <pthread.h>
-int main(void) {
-#ifndef __NetBSD__
-cpu_set_t s;
-CPU_ZERO(&s);
-pthread_setaffinity_np(0, sizeof(s), &s);
-#else
-cpuset_t *s = cpuset_create();
-cpuset_zero(s);
-pthread_setaffinity_np(0, cpuset_size(s), s);
-cpuset_destroy(s);
-#endif
-return 0;
-}" HAS_PTHREAD_SETAFFINITY_NP)
-CMAKE_POP_CHECK_STATE()
+set(CMAKE_THREAD_PREFER_PTHREAD TRUE)
+set(THREADS_PREFER_PTHREAD_FLAG TRUE)
+find_package(Threads REQUIRED)
```

Comment:
It is already 2022 and I agree that we can require a newer version of CMake than 2.8.12. OTOH, I'm not sure if the existing behavior is preserved with these changes. Old code detects the existence `pthread_setaffinity_np` and reports that using variable `HAS_PTHREAD_SETAFFINITY_NP`.

In case we need to continue using the existing approach, I think the fix for #2933 would be to use `CMAKE_C_SOURCE_RUNS` instead of -COMPILES, or set `-Werror,-Wimplicit-function-declaration` as CMAKE_C_FLAG.

Reply:
I'm very unfamiliar with cmake & cross-platform builds here, so I bow to your expertise here. The cmake version bump is because the pthreads check isn't available prior.